configuration/configuration_blackbox_test.go

```diff
@@ -31,7 +40,7 @@ func TestMain(m *testing.M) {
 
 func resetConfiguration() {
 	var err error
-	config, err = configuration.GetConfigurationData()
+	config, err = configuration.NewConfigurationData("../config.yaml")
```

Comment:
Thanks @alexeykazakov 

> It will make difference if we change the defaults but won't update the config.yaml.
True. We however ensure during PR review that config.yaml should always be updated along with code defaults. But you have a fair point, that the code should handle the validation.

I have added a new commit https://github.com/almighty/almighty-core/pull/884/commits/a39f38bd80c833480635c44fed82d291fcc79bdb to ensure that existing tests still use the default values and not config values, and the new ones should use config file values.

I did so, because the intention of the PR is to demonstrate the fix for the config.yaml scenario only.

I would probably have a table driven test both the scenarios in a different PR, by using a test suite, somewhat like this: https://github.com/sbose78/almighty-core/blob/694382b409797b57feb3e2b6ccb053a5f3233519/configuration/configuration_test.go#L203